spring-ai-core/src/main/java/org/springframework/ai/splitter/TextSplitter.java

```diff
@@ -49,6 +64,13 @@ private List<Document> createDocuments(List<String> texts, Map<String, Object> m
 					.stream()
 					.collect(Collectors.toMap(e -> e.getKey(), e -> e.getValue()));
 				Document newDoc = new Document(chunk, metadataCopy);
+
+				if (this.copyContentFormatter) {
+					// Copies the parent content formatter to the chucks documents it was
+					// spelt into.
+					newDoc = newDoc.updateContentFormatter(formatters.get(i));
```

Comment:
multiple formatters indexed by List of strings seems unusual to me.

Reply:
It is because the `createDocuments` takes a list of `texts`, one per source document, and merged metadata of all source documents. We can not merge the content formatters, so we have to pass in a list of formatters (one per source document) that corresponds in order to the list of texts. 

We can simplify the createDocuments to take one document at a time, but will loose the metadata merging, or introduce an internal record class like `Cuhnk(String text, Map<String, Object> metadata, ContentFormatter formatter)`?

Reply:
Actually,  why de we merge the metadata from all parent documents? Shouldn't the chunked docs inherit only the metadata their own parent? Not a mixture of metadata of all different parents? 
Later can make sense if we can determine that all parent documents come from the same initial source for example. But, currently, we don't have a convention to reliably recognise this.